tests/src/Unit/GetMatchingFieldTest.php

```diff
@@ -0,0 +1,130 @@
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Tests\UnitTestCase;
+use Drupal\og\OgGroupAudienceHelper;
+
+/**
+ * Tests the OgGroupAudienceHelper::getMatchingField method.
+ *
+ * @group og
+ *
+ * @coversDefaultClass \Drupal\og\OgGroupAudienceHelper
+ */
+class GetMatchingFieldTest extends UnitTestCase {
+
+
+  /**
+   * The entity type used for testing.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeInterface|\PHPUnit_Framework_MockObject_MockObject
+   */
+  protected $entityType;
+
+  /**
+   * The entity manager used for testing.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface|\PHPUnit_Framework_MockObject_MockObject
+   */
+  protected $entityManager;
+
+  /**
+   * The ID of the type of the entity under test.
+   *
+   * @var string
+   */
+  protected $entityTypeId;
+
+  /**
+   * The ID of the type of the bundle under test.
+   *
+   * @var string
+   */
+  protected $bundleId;
+
+  /**
+   * The group content entity.
+   *
+   * @var \Drupal\Core\Entity\ContentEntityInterface|\PHPUnit_Framework_MockObject_MockObject
+   */
+  protected $groupContent;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    $field_name = 'test_field';
+
+    $this->entityTypeId = $this->randomMachineName();
+    $this->bundleId = $this->randomMachineName();
+    $this->provider = $this->randomMachineName();
+
+    $field_storage_definition_prophecy = $this->prophesize(FieldStorageDefinitionInterface::class);
+    $field_storage_definition_prophecy->getCardinality()
+      ->willReturn(FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED)
+      ->shouldBeCalled();
+
+    $field_definition_prophecy = $this->prophesize(FieldDefinitionInterface::class);
+    $field_definition_prophecy->getFieldStorageDefinition()
+      ->willReturn($field_storage_definition_prophecy->reveal())
+      ->shouldBeCalled();
+
+    $field_definition_prophecy->getType()
+      ->willReturn('og_membership_reference')
+      ->shouldBeCalled();
+
+    $field_definition_prophecy->getSetting('target_type')
+      ->willReturn($this->entityTypeId)
+      ->shouldBeCalled();
+
+
+    $this->groupContent = $this->prophesize(ContentEntityInterface::class);
+
+    $this->groupContent->getFieldDefinition($field_name)
+      ->willReturn($field_definition_prophecy->reveal());
+
+    $this->groupContent->bundle()
+      ->shouldBeCalled();
+    $this->groupContent->getEntityTypeId()
+      ->shouldBeCalled();
+
+    // If the cardinality is unlimited getting a count of the field items is
+    // never expected, so just check it's not called.
+    $this->groupContent->get($field_name)
+      ->shouldNotBeCalled();
+
+    $this->entityType = $this->getMock('\Drupal\Core\Entity\EntityTypeInterface');
+    $this->entityType->expects($this->any())
+      ->method('getProvider')
+      ->will($this->returnValue($this->provider));
+
+    $this->entityManager = $this->getMock('\Drupal\Core\Entity\EntityTypeManagerInterface');
```

Comment:
@pfrenssen maybe you will notice what am I'm doing wrong here with the mocking?

When running test, `./vendor/phpunit/phpunit/phpunit  -c ./core/phpunit.xml.dist ./modules/og/tests/src/Unit/GetMatchingFieldTest.php`

I get:

```
PHPUnit 4.8.11 by Sebastian Bergmann and contributors.

F

Time: 260 ms, Memory: 7.75Mb

There was 1 failure:

1) Drupal\Tests\og\Unit\GetMatchingFieldTest::testGetMatchingField
Expectation failed for method name is equal to <string:getDefinition> when invoked zero or more times
Parameter 0 for invocation Drupal\Core\Entity\EntityTypeManagerInterface::getDefinition(null, true) does not match expected value.
Failed asserting that null matches expected 'tjm8jtse'.

/Library/WebServer/Documents/d8_dev/modules/og/src/OgGroupAudienceHelper.php:152
/Library/WebServer/Documents/d8_dev/modules/og/src/OgGroupAudienceHelper.php:94
/Library/WebServer/Documents/d8_dev/modules/og/tests/src/Unit/GetMatchingFieldTest.php:127
```

Unit testing isn't trivial...